tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3095,110 +3091,206 @@ LogicalResult ConvertTFLFakeQuantOp::matchAndRewrite(
+  DEF_FUNCTION_BINARY(LogicalOr);
+  DEF_FUNCTION_BINARY(Pow);
+
+  DEF_FUNCTION(TFLRelu);
+  DEF_FUNCTION(TFLRelu6);
+  DEF_FUNCTION(TFLEqual);
+  DEF_FUNCTION(TFLNotEqual);
+  DEF_FUNCTION(TFLGreater);
+  DEF_FUNCTION(TFLGreaterEqual);
+  DEF_FUNCTION(TFLAdd);
+  DEF_FUNCTION(TFLSub);
+  DEF_FUNCTION(TFLMul);
+  DEF_FUNCTION(TFLSquare);
+  DEF_FUNCTION(TFLSquaredDifference);
+  DEF_FUNCTION(TFLRound);
+  DEF_FUNCTION(TFLDiv);
+  DEF_FUNCTION(TFLMaximum);
+  DEF_FUNCTION(TFLMinimum);
+  DEF_FUNCTION(TFLFloorMod);
+  DEF_FUNCTION(TFLFloorDiv);
+  DEF_FUNCTION(TFLAddN);
+  DEF_FUNCTION(TFLAveragePool2D);
+  DEF_FUNCTION(TFLMaxPool2D);
+  DEF_FUNCTION(TFLConcatenation);
+  DEF_FUNCTION(TFLReshape);
+  DEF_FUNCTION(TFLRank);
+  DEF_FUNCTION(TFLShape);
+  DEF_FUNCTION(TFLExpandDims);
+  DEF_FUNCTION(TFLSqueeze);
+  DEF_FUNCTION(TFLFill);
+  DEF_FUNCTION(TFLElu);
+  DEF_FUNCTION(TFLSoftmax);
+  DEF_FUNCTION(TFLLogSoftmax);
+  DEF_FUNCTION(TFLSqrt);
+  DEF_FUNCTION(TFLL2Normalization);
+  DEF_FUNCTION(TFLReduceAny);
+  DEF_FUNCTION(TFLReduceMax);
+  DEF_FUNCTION(TFLReduceMin);
+  DEF_FUNCTION(TFLMean);
+  DEF_FUNCTION(TFLReduceProd);
+  DEF_FUNCTION(TFLSum);
+  DEF_FUNCTION(TFLConv2D);
+  DEF_FUNCTION(TFLTransposeConv);
+  DEF_FUNCTION(TFLDepthwiseConv2D);
+  DEF_FUNCTION(TFLFullyConnected);
+  DEF_FUNCTION(TFLBatchMatMul);
+  DEF_FUNCTION(TFLSplit);
+  DEF_FUNCTION(TFLSplitV);
+  DEF_FUNCTION(TFLPack);
+  DEF_FUNCTION(TFLUnpack);
+  DEF_FUNCTION(TFLTranspose);
+  DEF_FUNCTION(TFLTile);
+  DEF_FUNCTION(TFLSlice);
+  DEF_FUNCTION(TFLStridedSlice);
+  DEF_FUNCTION(TFLHardSwish);
+  DEF_FUNCTION(TFLZerosLike);
+  DEF_FUNCTION(TFLLess);
+  DEF_FUNCTION(TFLLessEqual);
+  DEF_FUNCTION(TFLPad);
+  DEF_FUNCTION(TFLPadV2);
+  DEF_FUNCTION(TFLResizeBilinear);
+  DEF_FUNCTION(TFLResizeNearestNeighbor);
+  DEF_FUNCTION(TFLSelect);
+  DEF_FUNCTION(TFLSelectV2);
+  DEF_FUNCTION(TFLSpaceToBatchNd);
+  DEF_FUNCTION(TFLBatchToSpaceNd);
+  DEF_FUNCTION(TFLSpaceToDepth);
+  DEF_FUNCTION(TFLDepthToSpace);
+  DEF_FUNCTION(TFLLogistic);
+  DEF_FUNCTION(TFLTanh);
+  DEF_FUNCTION(TFLPRelu);
+  DEF_FUNCTION(TFLLeakyRelu);
+  DEF_FUNCTION(TFLNeg);
+  DEF_FUNCTION(TFLYield);
+  DEF_FUNCTION(TFLCustom);
+  DEF_FUNCTION(TFLReverseV2);
+  DEF_FUNCTION(TFLQuantize);
+  DEF_FUNCTION(TFLDequantize);
+  DEF_FUNCTION(TFLConst);
+  DEF_FUNCTION(TFLQConst);
+  DEF_FUNCTION(TFLGather);
+  DEF_FUNCTION(TFLGatherNd);
+  DEF_FUNCTION(TFLSparseToDense);
+  DEF_FUNCTION(Constant);
+  DEF_FUNCTION(TFLOneHot);
+  DEF_FUNCTION(TFLArgMax);
+  DEF_FUNCTION(TFLFakeQuant);
 }
 
 // Creates an instance of the TensorFlow Lite dialect LegalizeTFL pass.
-std::unique_ptr<OperationPass<FuncOp>> createLegalizeTFLPass() {
-  return std::make_unique<LegalizeTFL>();
+std::unique_ptr<OperationPass<FuncOp>> createLegalizeTFLPass(
+  std::unordered_map<std::string, bool> legalization_enable)
+{
+  auto p = std::make_unique<LegalizeTFL>();
+  p->legalization_enable = legalization_enable;
```

Comment:
This needs to be a construction variable, we avoid mutation on passes.

Reply:
Done